### PR TITLE
Removing magic (`ucfirst`) and add CONST for constraint types

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Schema/Schema.php [hunk 1]
```diff
@@ -956,4 +955,66 @@ public function getSchemaCache(): SchemaCache
     {
         return $this->schemaCache;
     }
+
+    /**
+     * This method will call a `'loadTable' . ucfirst($type)` named method with the table name to obtain the metadata.
```
Comment (samdark) [inline src/Schema/Schema.php:960]: This description is no longer true.

**Comment by samdark** [comment]
Great idea making it more explicit. That exact method stood in my way of IDE-powered refactoring multiple times in Yii2.

**Comment by darkdef** [comment]
> 

I wanted use 'match'. But we have php 7.4

**Comment by vjik** [comment]
> I wanted use 'match'. But we have php 7.4

Maybe increase php to 8.0 in DB-packages? I think nobody uses and will not add them to existsing projects. New projects will be used PHP 8.0+.

File: src/Schema/Schema.php [hunk 2]
```diff
@@ -956,4 +955,66 @@ public function getSchemaCache(): SchemaCache
...
+     *
+     * @return mixed
+     */
+    protected function loadTableType(string $type, string $name)
+    {
+        switch ($type) {
+            case SchemaInterface::SCHEMA:
+                return $this->loadTableSchema($name);
+            case SchemaInterface::PRIMARY_KEY:
+                return $this->loadTablePrimaryKey($name);
```
Comment (vjik) [inline src/Schema/Schema.php:973]: Maybe in this PR add these and other methods to `SchemaInterface` ?

File: src/Schema/Schema.php (hunk 1 above)
Comment (darkdef) [reply src/Schema/Schema.php]: changed

File: src/Schema/Schema.php (hunk 2 above)
Comment (darkdef) [reply src/Schema/Schema.php:920]: All methods added to SchemaInterface

File: src/Schema/SchemaInterface.php [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+    public const TYPE_BOOLEAN = 'boolean';
+    public const TYPE_MONEY = 'money';
+    public const TYPE_JSON = 'json';
+
+    public function createQueryBuilder(): QueryBuilder;
+
+    /**
+     * @return QueryBuilder the query builder for this connection.
+     */
+    public function getQueryBuilder(): QueryBuilder;
```
Comment (samdark) [inline src/Schema/SchemaInterface.php:58]: Do we need separate `createQueryBuilder()` and `getQueryBuilder()`?

File: src/Schema/SchemaInterface.php [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+    public function supportsSavepoint(): bool;
+
+    /**
+     * Creates a new savepoint.
+     *
+     * @param string $name the savepoint name
+     *
+     * @throws Exception|InvalidConfigException|Throwable
+     */
+    public function createSavepoint(string $name): void;
```
Comment (samdark) [inline src/Schema/SchemaInterface.php:172]: Not sure it belongs to schema. That's active action and schema is about getting DDL.

File: src/Schema/SchemaInterface.php [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+     * Executes the INSERT command, returning primary key values.
+     *
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column data (name => value) to be inserted into the table.
+     *
+     * @throws Exception|InvalidCallException|InvalidConfigException|Throwable
+     *
+     * @return array|false primary key values or false if the command fails.
+     */
+    public function insert(string $table, array $columns);
```
Comment (samdark) [inline src/Schema/SchemaInterface.php:217]: Doesn't seem to belong to schema.

File: src/Schema/SchemaInterface.php [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+     *
+     * This can be one of {@see Transaction::READ_UNCOMMITTED}, {@see Transaction::READ_COMMITTED},
+     * {@see Transaction::REPEATABLE_READ} and {@see Transaction::SERIALIZABLE} but also a string containing DBMS
+     * specific syntax to be used after `SET TRANSACTION ISOLATION LEVEL`.
+     *
+     * @throws Exception|InvalidConfigException|Throwable
+     *
+     * {@see http://en.wikipedia.org/wiki/Isolation_%28database_systems%29#Isolation_levels}
+     */
+    public function setTransactionIsolationLevel(string $level): void;
```
Comment (samdark) [inline src/Schema/SchemaInterface.php:205]: Doesn't seem to belong to schema.

Comment (darkdef) [reply src/Schema/SchemaInterface.php:205]: it's be fixed in next PR

File: src/Schema/SchemaInterface.php (hunk 5 above)
Comment (darkdef) [reply src/Schema/SchemaInterface.php:217]: it's be fixed in next PR

File: src/Schema/SchemaInterface.php (hunk 4 above)
Comment (darkdef) [reply src/Schema/SchemaInterface.php:172]: it's be fixed in next PR

File: src/Schema/SchemaInterface.php (hunk 3 above)
Comment (darkdef) [reply src/Schema/SchemaInterface.php:59]: it's be fixed in next PR

**Review by terabytesoftw — APPROVED** [review]

**Pull request merged**